File: maths/L07Password.js
var chars = ['a', 'b', 'c', 'd', 'e']
var result = []

function getPassword(passwordChars, num, password) {
  if (num == 0) {
    return result.push(password)
  } else {
    for (var i = 0; i < passwordChars.length; i++) {
      getPassword(passwordChars, num-1, password+passwordChars[i])
    }
  }
}

getPassword(chars, 4, '')

File: maths/L07Password.py
import copy

password = 'bacdcd'
classes = ['a', 'b', 'c','d', 'e']

def get_password(n, result = ''):
    if n == 0:
        if result == password:
            print(password)
    else:
        for i in classes:
            new_result = copy.copy(result)
            new_result = new_result + i
            get_password(n-1, new_result)

get_password(6)
